### PR TITLE
Use caches for getPolyfillMeta, listAliases, and listPolyfills

### PR DESCRIPTION
**Pull request by JakeChampion**
Resolves https://github.com/Financial-Times/polyfill-library/issues/147
Resolves https://github.com/Financial-Times/polyfill-library/issues/148

### CODE REVIEW

File: lib/sources.js [hunk 1]
```diff
@@ -13,31 +15,41 @@ const polyfillDirectory = path.join(__dirname, "../polyfills/__dist");
...
 
 /**
  * Get a list of all the polyfill aliases which exist within the collection of polyfill sources.
  * @returns {Promise<Array>} A promise which resolves with an object of all the polyfill aliases within the collection.
  */
-function listAliases() {
-	return readFile(path.join(polyfillDirectory, "aliases.json")).then(JSON.parse);
-}
+const listAliases = (function() {
+	const aliases = readFile(path.join(polyfillDirectory, "aliases.json")).then(JSON.parse);
```
Comment (bringking) [inline lib/sources.js]: Would be awesome to cache this one as well, it gets read from disk on every request as well

**Comment by bringking** [comment]
@JakeChampion, this is great. Thank you! Any particular reason you want to avoid using require? Just curious

**Comment by JakeChampion** [comment]
@bringking avoiding require is because it is a blocking call if it is the first time the file is being requied. If polyfill-library is used on a HTTP server and that blocking call happened, it would stop other requests from being handled at the same moment in time.

File: lib/sources.js (hunk 1 above)
Comment (JakeChampion) [reply lib/sources.js]: Thanks for the review :-)

**Pull request merged**